### PR TITLE
Add an e2e for ignite run

### LINKED ISSUE #391: Write e2e test cases for ignite run

### PR DESCRIPTION
**Pull request by chanwit**
Added a simple e2e test for `ignite run` command. This one is to test under docker / docker-bridge environment. Tests for containerd / cni will be added after #411 get merged.

e2e tests will be used locally at the moment for devs, and later should be with the CI when we could test them in a remote KVM environment.

This PR fixes #391

### CODE REVIEW

**Comment by chanwit** [comment]
kindly ping @stealthybox if this e2e looks make sense to you, I'll continue adding more tests.

**Comment by chanwit** [comment]
Please merge #413 before this one.

File: e2e/run_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+//
+// This is the e2e package to run tests for Ignite
+// Currently, we do local e2e tests only
+// we have to wait until the CI setup to allow Ignite to run with sudo and in a KVM environment.
+//
+// How to run tests:
+// sudo IGNITE_E2E_HOME=$PWD $(which go) test ./e2e/.
```
Comment (stealthybox) [inline e2e/run_test.go]: adding `-count 1` will disable caching
```suggestion
// sudo IGNITE_E2E_HOME=$PWD $(which go) test ./e2e/. -count 1
```

**Comment by chanwit** [comment]
Fixed per @stealthybox comments
For the `make tidy` issue, it should be OK for now as I rebased this PR on top of other changes.

File: Makefile [hunk 2]
```diff
@@ -228,3 +228,6 @@ test-docs: build-docs
 serve-docs: build-docs
 	@echo Stating docs website on http://localhost:${DOCS_PORT}/_build/html/index.html
 	@docker run -i --rm -p ${DOCS_PORT}:8000 -e USER_ID=$$UID ignite-docs
+
+e2e: $(BINARIES)
```
Comment (stealthybox) [inline Makefile]: This might need to be `build-all` instead, because we should be testing with a fresh OCI image for `ignite-spawn`

Comment (chanwit) [reply Makefile]: oh right. this should be fix to cover the case when `ignite-spawn` is changed.
thank you @stealthybox !

**Pull request merged**